w3-filecoin.md

```diff
@@ -102,198 +105,316 @@ For example, an Aggregator can authorize invocations from `did:web:web3.storage`
+    Aggregator-->>Storefront: Receipt with inclusion proof
+    end
+    Aggregator-->>Storefront: Receipt akwnoledging offer
     deactivate Aggregator
-    Aggregator-->>Storefront: receipt issued <br>with inclusion proof
 ```
 
 ### Aggregator offers dealer an aggregate
 
-When the Aggregator has enough content pieces to build a qualified aggregate (dealers MAY impose different requirements), it MUST submit a Filecoin deal for the aggregate to a Dealer. Dealer MUST issue signed receipt acknowledging submission, actual deal negotiation with Filecoin Storage Providers MAY carry out of band.
+When the Aggregator has enough content pieces to build a qualified aggregate (dealers MAY impose different requirements), it MUST offer an aggregate to the Dealer. Dealer MUST issue signed receipt acknowledging an offer, actual deal negotiation with Filecoin Storage Providers MAY be carried out of band.
 
-Once a Dealer receives an aggregate offer it is queued for negotiations with Storage Providers. Issued receipt is a proofs transition of the (offered aggregate) state from `uninitialized` into `pending`. If Dealer receives request with an aggregate already in pipeline it MUST simply reissue receipt with a same result and effects as the original request.
+If Dealer receives a request with an aggregate multiple times it MUST (re)issue receipt with the same result and effects.
 
-Issued receipt MUST link to a followup task (using `.fx.join` field) that either succeeds (if the aggregate deal made it into Filecoin chain) or fails (e.g. if Storage Provider failed to replicate and reported an error) so that its receipt COULD be looked up by it.
+> ℹ️ Invocation nonce could be used to force aggregate to be reprocessed.
 
-After a Dealer dequeues the aggregate, it will interact with available Filecoin Storage Providers, in order to establish a previously determined (out of band) number of deals. Depending on storage providers availability, as well as the content present in the offer, the aggregate MAY be handled or not. A receipt is created to proof the transition of the aggregate state from `pending` into `done`.
+Issued receipt MUST have `fx.join` [effect] linking to a `aggregate/accept` task which either succeeds with filecoin [`DataAggregationProof`] result or fails (e.g. if Storage Provider failed to replicate and reported an error).
 
-> Note: Dealer MAY have several intermediate steps and states it transitions through, however those intentionally are not captured by this protocol, because storefront will take no action until success / failure condition is met.
+Dealer MUST broker deal(s) with Filecoin Storage Providers (out of band). It MUST issue receipt for `aggregate/accept` task with succeed or failure result depending on the availability of storage providers and their ability to replicate content pieces in the aggregate. Successful task MUST have [`DataAggregationProof`] as result and contain no [effect]s.
```

Comment:
There is something weird with the start of this sentence. Could not understand exactly, but maybe `broker` should be replaced by `handle`?

Reply:
Hmm, I think I like verb broker better because it is more specific which per google means

<img width="675" alt="image" src="https://github.com/web3-storage/specs/assets/21236/edb22768-8a71-48b3-bf6a-2b63487e158e">


If you don't like **broker** we could change it to **arrange** instead.

Reply:
That seems good. Wow, did not know about this verb 😂